### PR TITLE
Clean-up and fix VisualTreeElementExtensions hit testing

### LINKED ISSUE #4613: VisualTreeElementExtensions.GetVisualTreeElements does not work for transformed elements
### Description

Hit testing [VisualTreeElementExtensions.GetVisualTreeElements](https://github.com/dotnet/maui/blob/cd3a8193fefb2d1e18993e0f15446989fa74a6cc/src/Core/src/Core/Extensions/VisualTreeElementExtensions.cs#L64) does not work well for transformed elements.
MauiHitTestIssue

### Steps to Reproduce

1. Create new WinUI/MAUI app.
2. Open MainPage.xaml
3. Add Scale="2" to <Button> element.
4. Build and run.
5. Use  [VisualTreeElementExtensions.GetVisualTreeElements](https://github.com/dotnet/maui/blob/cd3a8193fefb2d1e18993e0f15446989fa74a6cc/src/Core/src/Core/Extensions/VisualTreeElementExtensions.cs#L64)  to hit test Button
 [VisualTreeElementExtensions.GetVisualTreeElements](https://github.com/dotnet/maui/blob/cd3a8193fefb2d1e18993e0f15446989fa74a6cc/src/Core/src/Core/Extensions/VisualTreeElementExtensions.cs#L64) : can hit test button at any corner
**Actual**: hit testing near bottom or right edges does not find button
**Note**: similar issue with Rotate transform and possibly others

### Version with bug

Preview 13

### Last version that worked well

Unknown/Other

### Affected platforms

I was *not* able test on other platforms

### Affected platform versions

Windows

### Did you find any workaround?

no

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by antonfirsov**
### Description of Change

- WinUI: Fix rectangle hit testing by always delegating it to platform code
- Also improve logic on other platforms, though I wasn't able to test it
- Remove the `usePlatformViewBounds` argument from the API-s
- Instead of `IList<IVisualTreeElement>`, return `IReadOnlyList<IVisualTreeElement>` from all `VisualTreeElementExtensions` methods
- Add HitTestingPage to the Samples app

HitTesting

### Issues Fixed

Fixes #4613

### CODE REVIEW

File: src/Controls/samples/Controls.Sample/Maui.Controls.Sample.csproj [hunk 1]
```diff
@@ -52,8 +52,21 @@
...
+      <SubType>Code</SubType>
+      <DependentUpon>HitTestingPage.xaml</DependentUpon>
+    </Compile>
+  </ItemGroup>
+
   <Import Condition=" '$(UseMaui)' != 'true' " Project="..\..\..\BlazorWebView\src\Maui\build\Microsoft.AspNetCore.Components.WebView.Maui.targets" />
 
   <Import Project="../../../Maui.InTree.props" Condition=" '$(UseMaui)' != 'true' " />
 
+  <ItemGroup>
```
Comment (jsuarezruiz) [inline src/Controls/samples/Controls.Sample/Maui.Controls.Sample.csproj:66]: Are not necessary changes although we can clean the csproj later as well.

**Review by jsuarezruiz — APPROVED** [review]

File: src/Controls/samples/Controls.Sample/Maui.Controls.Sample.csproj (hunk 1 above)
Comment (antonfirsov) [reply src/Controls/samples/Controls.Sample/Maui.Controls.Sample.csproj]: Fixed in a1a7afd

File: src/Core/src/Core/Extensions/VisualTreeElementExtensions.cs [hunk 2]
```diff
@@ -52,70 +53,64 @@ static IList<IVisualTreeElement> GetVisualTreeDescendantsInternal(this IVisualTr
 		}
 
 		/// <summary>
-		/// Gets list of a Visual Tree Elements children based off of a given x, y point.
+		/// Gets list of a Visual Tree Elements children based off of a rectangle defined by it's coordinates.
```
Comment (drasticactions) [inline src/Core/src/Core/Extensions/VisualTreeElementExtensions.cs:56]: nit: "its coordinates"

**Review by drasticactions — APPROVED** [review]
LGTM, I wanna get @etvorun's view before we go ahead if at all possible.

This shouldn't break our usage of the APIs (we were always using the platform view bounds and were not placed in the bool, so it should keep working). This should be good.

File: src/Core/src/Core/Extensions/VisualTreeElementExtensions.cs (hunk 2 above)
Comment (etvorun) [reply src/Core/src/Core/Extensions/VisualTreeElementExtensions.cs:56]: Please update <summary/> comment to specify that coordinates are in platform units (as opposite to pixels). For all methods which take positional coordinates.

**Review by etvorun — APPROVED** [review]

File: src/Core/src/Core/Extensions/VisualTreeElementExtensions.cs (hunk 2 above)
Comment (Redth) [inline src/Core/src/Core/Extensions/VisualTreeElementExtensions.cs:56]:
```suggestion
		/// Gets list of a Visual Tree Elements children based off of a rectangle defined by its coordinates which are specified in platform units, not pixels.
```

Comment (etvorun) [reply src/Core/src/Core/Extensions/VisualTreeElementExtensions.cs]: We need such comment for all public methods

**Pull request merged**